cuda_core/cuda/core/experimental/_program.py

```diff
@@ -378,6 +379,23 @@ def __init__(self, code, code_type, options: ProgramOptions = None):
                 raise TypeError("c++ Program expects code argument to be a string")
             # TODO: support pre-loaded headers & include names
             # TODO: allow tuples once NVIDIA/cuda-python#72 is resolved
+
+            supported_archs = handle_return(nvrtc.nvrtcGetSupportedArchs())
+
+            if options is not None:
+                arch_not_supported = (
+                    options.arch is not None and int(options.arch.split("_")[-1]) not in supported_archs
+                )
+                default_arch_not_supported = (
+                    options.arch is None
```

Comment:
This is no longer possible now that we default to the current arch when creating ProgramOptions without arch being set, so this logic here and below can be removed/simplified.